pkgs/applications/audio/faust/faust2jaqt.nix

```diff
@@ -17,10 +19,35 @@ faust.wrapWithBuildEnv {
 
   propagatedBuildInputs = [
     jack2
-    qt4
+    qtbase
     libsndfile
     alsa-lib
     which
   ];
 
+  dontWrapQtApps = true;
+
+  # Wrap the binary coming out of the the compilation script, so it knows QT_PLUGIN_PATH
+  wrapBinary = writeText "wrapBinary" ''
+    source ${makeWrapper}/nix-support/setup-hook
+    for p in $FILES; do
+      workpath=$(pwd)
+      cd -- "$(dirname "$p")"
+      binary=$(basename --suffix=.dsp "$p")
+      mv $binary ."$binary"-wrapped
+      makeWrapper ./."$binary"-wrapped ."$binary"-temp  --set QT_PLUGIN_PATH "${qtbase}/${qtbase.qtPluginPrefix}"
+      mv ."$binary"-temp $binary
+      sed -i $binary -e 's@exec@cd "$(dirname "$(readlink -f "''${BASH_SOURCE[0]}")")" \&\& exec@g'
+      cd $workpath
+    done
+  '';
+  # append the wrapping code to the compilation script (cat line)
+  # prevent the qmake error when running the script (sed line)
+  preFixup = ''
+    for script in "$out"/bin/*; do
+      echo $script
+      cat ${wrapBinary} >> $script
+      sed -i "/QMAKE=/c\ QMAKE="${qtbase.dev}/bin/qmake"" $script
+    done
```

Comment:
```suggestion
  preFixup = ''
    for script in "$out"/bin/*; do
      # append the wrapping code to the compilation script
      echo "${wrapBinary}" >> $script
      # prevent the qmake error when running the script
      sed -i "/QMAKE=/c\ QMAKE="${qtbase.dev}/bin/qmake"" $script
    done
```

Reply:
Thanks for the review!
When I try these suggestions regarding wrapping, I end up with:
```bash
source /nix/store/7w5cmms8sh5pv70xgz2ysmhgdi12nwyf-make-shell-wrapper-hook/nix-support/setup-hook
for p in ; do
  workpath=/build/source
  cd -- .
  binary=
  mv  .-wrapped
  makeWrapper ./.-wrapped .-temp  --set QT_PLUGIN_PATH /nix/store/rb7gbnldym6dphxam2znp9kiw3flzw8n-qtbase-5.15.8-bin/lib/qt-5.15.8/plugins
  mv .-temp
  sed -i  -e 's@exec@cd /nix/store/c3f4jdwzn8fm9lp72m91ffw524bakp6v-stdenv-linux \&\& exec@g'
  cd
done
```
While I need:
```bash
source /nix/store/7w5cmms8sh5pv70xgz2ysmhgdi12nwyf-make-shell-wrapper-hook/nix-support/setup-hook
for p in $FILES; do
  workpath=$(pwd)
  cd -- "$(dirname "$p")"
  binary=$(basename --suffix=.dsp "$p")
  mv $binary ."$binary"-wrapped
  makeWrapper ./."$binary"-wrapped ."$binary"-temp  --set QT_PLUGIN_PATH "/nix/store/rb7gbnldym6dphxam2znp9kiw3flzw8n-qtbase-5.15.8-bin/lib/qt-5.15.8/plugins"
  mv ."$binary"-temp $binary
  sed -i $binary -e 's@exec@cd "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")" \&\& exec@g'
  cd $workpath
done
```

Reply:
That leaves me with:
```bash
source /nix/store/l84drzijhk1ilnn3yvv2r42wh2lj43ni-make-shell-wrapper-hook/nix-support/setup-hook
for p in $FILES; do
  workpath=$PWD
  cd -- "$(dirname "$p")"
  binary=$(basename --suffix=.dsp "$p")
  mv $binary ."$binary"-wrapped
  makeWrapper "./.$binary-wrapped" ".$binary-temp" --set QT_PLUGIN_PATH "/nix/store/rzp9gihz7rfg9l59gjpzavy2wrg7vjdj-qtbase-5.15.9-bin/lib/qt-5.15.9/plugins"
  mv ".$binary-temp" "$binary"
  sed -i $binary -e s@exec@cd /nix/store/smcpmkfc7bx90ry7wmr6nckg83anjj0k-stdenv-linux && exec@g
  cd $workpath
done
```

Reply:
So only the bash source part is wrong. If you can't solve that on your own, we can also keep it like it is, to make it less brittle on changes. You decide.

Reply:
I think the source part is OK, it's mainly the end that is wrong.
I need:
```bash
source /nix/store/7w5cmms8sh5pv70xgz2ysmhgdi12nwyf-make-shell-wrapper-hook/nix-support/setup-hook
for p in $FILES; do
  workpath=$(pwd)
  cd -- "$(dirname "$p")"
  binary=$(basename --suffix=.dsp "$p")
  mv $binary ."$binary"-wrapped
  makeWrapper ./."$binary"-wrapped ."$binary"-temp  --set QT_PLUGIN_PATH "/nix/store/rb7gbnldym6dphxam2znp9kiw3flzw8n-qtbase-5.15.8-bin/lib/qt-5.15.8/plugins"
  mv ."$binary"-temp $binary
  sed -i $binary -e 's@exec@cd "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")" \&\& exec@g'
  cd $workpath
done
```